### DISCUSSION TITLE
Hybrid rendering and dynamic data from cookie

### DISCUSSION THREAD

**Discussion by lewebsimple** [nuxt/nuxt#19175] [Questions] [answered]
I'm trying to configure hybrid rendering on Vercel for a headless Shopify e-commerce site and I'm having trouble getting the cart item count to display correctly on pre-rendered pages.

The cart data comes from a composable that executes the following:
```
  const cartCookie = useCookie<CartFragment>("cart");
  const cart = useState<CartFragment>("cart", () => cartCookie.value || cartDefault);
  watch(cart, (newCart) => (cartCookie.value = newCart));
```

I tried wrapping the cart icon component inside `<ClientOnly>` but the cart item count is always stale when the cache is hit.

Is the only / best solution to initialize the cart state inside `onMounted` or am I missing something obvious ?

**Comment by Baroshem** [comment, marked as answer]
Hey,

Yes, I would recommend that. Cart is one of the things in e-commerce that are user-related data that you should have only on the frontend. Initialization as well. Otherwise you might run into situations where you could by mistake render a cart for user A with content from user B by for example serving content from CDN.

In Summary, I would recommend to initialize the cart state inside `onMounted` :)

**Reply by lewebsimple** [reply]
Thanks, this makes sense. I was overthinking this trying to find a pure "nuxty" way of achieving this.
